pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -461,3 +469,32 @@ func findNextFreeNumber(set map[int32]bool, baseValue int32) *int32 {
 	set[baseValue] = true
 	return &baseValue
 }
+
+func getLatestSku(ctx context.Context, opt *Options, factory azureclient.Factory) (string, error) {
```

Comment:
I would honestly restructure the code a little bit here. Instead of "softcoding" the image by using the `vmImage` function directly, maybe modify the function signature so that it accepts a `compute.ImageReference`. So the function would indeed fetch the latest SKU for the passed image.

Reply:
@kon-angelo https://github.com/gardener/gardener-extension-provider-azure/pull/668/commits/a5ea43e5a18bb30feebe9982b9fca82f2fc572a8 committed, `vmImage` removed then I change to return `compute.ImageReference` from `getLatestSku`